src/jit/lowerxarch.cpp

```diff
@@ -2299,6 +2299,18 @@ void Lowering::TreeNodeInfoInitLogicalOp(GenTree* tree)
         // as reg optional.
         SetRegOptionalForBinOp(tree);
     }
+
+    if (varTypeIsByte(op1))
+    {
+        regMaskTP regMask;
+        bool      containedNode = (op2->gtLsraInfo.srcCount == 0) && (op2->gtLsraInfo.dstCount == 0);
+        if (!containedNode)
+        {
+            regMask = op2->gtLsraInfo.getSrcCandidates(l);
+            assert(regMask != RBM_NONE);
+            op2->gtLsraInfo.setSrcCandidates(l, regMask & ~RBM_NON_BYTE_REGS);
+        }
```

Comment:
> 2) Locals that are normalized on store, should have their type normalized to int or long by importer.

I think that the condition here is actually "lclVars that are not normalized on load": AFAICT the importer never calls `lvNormalizeOnStore`.

Incidentally, does anybody know why args are considered normalize-on-load but not normalize-on-store? I can understand the former--I assume that we do load normalization to guard against the caller not doing store-normalization--but I'm a bit lost as to the former (unless the arg is also aliasable).